app/controllers/events_controller.rb

```diff
@@ -1,5 +1,6 @@
 class EventsController < ApplicationController
   before_action :set_event, only: %i[update destroy edit]
+  before_action :authenticate_user!, only: %i[new create edit update destroy]
```

Comment:
If you're gonna use it always, just do `beforr_action : authenticate_user!`

Reply:
users must be logged in to view public events or they can view these events without being logged in?

Reply:
But you're letting them see all events without beings logged in 
My first comment was wrong you're omitting the show action but that means every user can see all the events without being logged in

Reply:
yes, that's why I asked the previous ones, to see if we leave it as before, that anyone can see public events without session or if the session is necessary to be able to see those events.